experimental/components/VideoPlayer/library/src/commonMain/kotlin/org/jetbrains/compose/videoplayer/VideoPlayer.kt

```diff
@@ -1,10 +1,112 @@
 package org.jetbrains.compose.videoplayer
 
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.listSaver
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+
+data class Progress(
+    val fraction: Float,
+    // TODO: Use kotlin.time.Duration when Kotlin version is updated.
+    //  See https://github.com/Kotlin/api-guidelines/issues/6
+    val timeMillis: Long
+)
 
 @Composable
-fun VideoPlayer(url: String, width: Int, height: Int) {
-    VideoPlayerImpl(url, width, height)
+fun VideoPlayer(
+    url: String,
+    state: VideoPlayerState,
+    modifier: Modifier = Modifier,
+    onFinish: (() -> Unit)? = null
+) = VideoPlayerImpl(
+    url = url,
+    isResumed = state.isResumed,
+    volume = state.volume,
+    speed = state.speed,
+    seek = state.seek,
+    isFullscreen = state.isFullscreen,
+    modifier = modifier,
+    onFinish = onFinish
+)
+
+internal expect fun VideoPlayerImpl(
+    url: String,
+    isResumed: Boolean,
+    volume: Float,
+    speed: Float,
+    seek: Float,
+    isFullscreen: Boolean,
+    modifier: Modifier,
+    onFinish: (() -> Unit)?
+): State<Progress>
```

Comment:
It's a bit unusual when a Composable function that draws/displays some content returns some value (state in this case)  at the same time.

I would pass progressState as a parameter and mutate it in the VideoPlayerImpl. 

Or maybe it's even better to the progress details to VideoPlayerState.

Reply:
Updated the code.